packages/@react-aria/list/src/ListGridKeyboardDelegate.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {GridCollection} from '@react-types/grid';
+import {GridKeyboardDelegate, GridKeyboardDelegateOptions} from '@react-aria/grid';
+import {Key} from 'react';
+import {Rect} from '@react-stately/virtualizer';
+
+export class ListGridKeyboardDelegate<T> extends GridKeyboardDelegate<T, GridCollection<T>> {
```

Comment:
Open to name suggestions

Reply:
What are the differences between this and ListKeyboardDelegate now?

Reply:
ListKeyboardDelegate will skip over disabled items whereas this one doesn't. Also handles the `getKeyLeftOf`/`getKeyRightOf` that useGridCell calls which ListKeyboardDelegate doesn't

Reply:
Sure I can go ahead and try modifying ListKeyboardDelegate instead

Reply:
I've done some poking around and have come up with https://github.com/adobe/react-spectrum/pull/3000/commits/f2ef4ef8e995f3e11f1520bfd35c9b450a74f86c, which leverages `ListLayout` as the keyboard delegate instead of the new `ListGridKeyboardDelegate`. I feel a bit iffy about it and it still has some finicky behavior that I can't resolve without bringing in some grid specific logic into ListLayout. 

Pain points:
- An effort has been made to make the tracked `focusedKey` always be a row key. However, due to the nature of `useGridCell`, it is possible for the `focusedKey` to be a cell key (e.g. if a user clicks on a row button), resulting in some broken behavior since the ListLayout wasn't designed to handle such a distinction. If I get rid of the `onFocus` from `useGridCell` that would fix the problem of `focusedKey` being set to a cell key but would then mean focus wouldn't update properly if the user clicks on said row buttons. Focus would also get highjacked from a menu button after closing the menu and returned onto the row since `useGridRow` doesn't have it's own dedicated `focus` func that handles these kinds of situations.  
- the list collection only has access to the `row` keys, not the `cell` keys. This makes using `ListKeyboardDelegate`/`ListLayout` tricky since they don't know what to do with the `cell` keys unless I modify those functions so they always try to grab the `row` key from the provided grid collection
- This mixture of using a grid collection with layouts/delegates that don't have a concept of `row` vs `cell` is pretty painful in general
- `ListKeyboarddDelegate` makes an assumption that the component isn't virtualized and uses query selectors to find row height information

I found the old approach w/ the new keyboard delegate to be a bit easier to navigate since it was extending off the GridKeyboardDelegate and I was able to ensure the `focusedKey` would always be the `cell` key meaning there were less pitfalls